tools/GSDumpGUI/Forms/frmMain.cs

```diff
@@ -315,34 +327,30 @@ private void cmdOpenIni_Click(object sender, EventArgs e)
             Process.Start(AppDomain.CurrentDomain.BaseDirectory + "GSDumpGSDXConfigs\\inis\\gsdx.ini");
         }
 
-        private void lstDumps_SelectedIndexChanged(object sender, EventArgs e)
+        private void UpdatePreviewImage(object sender, GsFiles<GsDumpFile>.SelectedIndexUpdatedEventArgs args)
         {
-            if (lstDumps.SelectedIndex != -1)
+            if (pctBox.Image != NoImage)
+                pctBox.Image?.Dispose();
+            if (_availableGsDumps.Selected?.PreviewFile == null)
             {
-                String [] Extensions = new String[] { ".png", ".bmp" };
-                String DumpFileName = lstDumps.SelectedItem.ToString().Split(new char[] { '|' })[0];
-                String Filename = Path.GetDirectoryName(Settings.DumpDir + "\\") + "\\" + Path.GetFileNameWithoutExtension(DumpFileName);
-
-                foreach (String Extension in Extensions)
-                {
-                    if (File.Exists(Filename + Extension))
-                    {
-                        pctBox.Load(Filename + Extension);
-                        pctBox.Cursor = Cursors.Hand;
-                        return;
-                    }
-                }
                 pctBox.Image = NoImage;
                 pctBox.Cursor = Cursors.Default;
             }
+            else
+            {
+                pctBox.Load(_availableGsDumps.Selected.PreviewFile.FullName);
+                pctBox.Cursor = Cursors.Hand;
+            }
+
+            pctBox.Tag = _availableGsDumps.Selected?.PreviewFile?.FullName;
         }
 
-        private void pctBox_Click(object sender, EventArgs e)
+        private static void PreviewImageClick(object sender, EventArgs e)
```

Comment:
Maybe just private void? Otherwise for me it fails to compile currently like this.

Reply:
No the `this` in the designer needs to be removed. I think git pull should contain the fix.

Reply:
Ah yea that did it.